CONTRIBUTING.md

```diff
@@ -0,0 +1,111 @@
+# Code Style Guidelines
+
+Most of rules is checked with help of the *maven-checkstyle-plugin* during the `validate` phase. 
+Checkstyle rules are more important than this document.
+
+## Resulting from long experience
+
+* To the largest extent possible, all fields shall be private. Use an IDE to generate the getters and setters.
+* If a class has more than one `volatile` member field, it is probable that there are subtle race conditions. Please consider where appropriate encapsulation of the multiple fields into an immutable value object replace the multiple `volatile` member fields with a single `volatile` reference to the value object (or perhaps better yet an `AtomicReference` to allow for `compareAndSet` - if compare-and-set logic is appropriate).
+* If it is `Serializable` it shall have a `serialVersionUID` field. Unless code has shipped to users, the initial value of the `serialVersionUID` field shall be `1L`.
+
+## Indentation
+
+1. **Use spaces.** Tabs are banned.
+2. **Java blocks are 4 spaces.** JavaScript blocks as for Java. **XML nesting is 4 spaces**
+
+## Field Naming Conventions
+
+1. "hungarian"-style notation is banned (i.e. instance variable names preceded by an 'm', etc)
+2. If the field is `static final` then it shall be named in `ALL_CAPS_WITH_UNDERSCORES`.
+3. Start variable names with a lowercase letter and use camelCase rather than under_scores.
+4. Spelling and abreviations: If the word is widely used in the JVM runtime, stick with the spelling/abreviation in the JVM runtime, e.g. `color` over `colour`, `sync` over `synch`, `async` over `asynch`, etc.
+5. It is acceptable to use `i`, `j`, `k` for loop indices and iterators. If you need more than three, you are likely doing something wrong and as such you shall either use full descriptive names or refactor.
+6. It is acceptable to use `e` for the exception in a `try...catch` block.
+7. You shall never use `l` (i.e. lower case `L`) as a variable name.
+
+## Line Length
+
+To the greatest extent possible, please wrap lines to ensure that they do not exceed 120 characters.
```

Comment:
150 doesn't mean that you should wrote every line with 150 symbols. Usually only logger is wide.